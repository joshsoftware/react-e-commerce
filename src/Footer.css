li{
    margin-top: 12px;
}
.footer{
background-color: black ;
color: white;
margin: 0 !important;
}