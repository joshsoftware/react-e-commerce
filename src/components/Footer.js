import React from 'react';
// import '../Footer.css'
import UlComponent from './UlComponent';
import LiComponent from './LiComponent';
import ContainerWrapper from './ContainerWrapper';
import ColumnWrapper from './ColumnWrapper';
import RowWrapper from './RowWrapper';
import FooterData from '../FooterData';

const Footer = () => {
  let arr = [];
  let arr1 = [];
  let arr2 = [];
  const categories = FooterData[0].categories;

  arr.push(<LiComponent key={'categories-element'} data="Categories" />);
  categories.map((category) => {
    return arr.push(
      <LiComponent key={category.id} data={<a href={category.url}>{category.name}</a>} />
    );
  });

  const partner = FooterData[1].partners;

  arr1.push(<LiComponent key={'partners-element'} data="Partners" />);
  partner.map((partner) => {
    return arr1.push(<LiComponent key={partner.id} data={partner.name} />);
  });

  const contacts = FooterData[2].contactus;

  arr2.push(<LiComponent key={'contact-us-details'} data="Contact Us" />);
  contacts.map((contact) => {
    return arr2.push(<LiComponent key={contact.id} data={contact.name} />);
  });

  let a1 = [];
  a1.push(
    <ColumnWrapper key={'column-categories'} data={<UlComponent liElement={arr} />} xs={4} />
  );
  a1.push(<ColumnWrapper key={'column-partners'} data={<UlComponent liElement={arr1} />} xs={4} />);
  a1.push(
    <ColumnWrapper key={'column-contact-us'} data={<UlComponent liElement={arr2} />} xs={4} />
  );
  a1.push(
    <div key={'footer-bootom'} className="footer-bottom">
      &copy;Copyright XYZ.ltd {new Date().getFullYear()}
    </div>
  );

  return (
    <footer className="bg-dark text-white">
      <ContainerWrapper data={<RowWrapper data={a1} />} fluid={true} />
    </footer>
  );
};
export default Footer;
