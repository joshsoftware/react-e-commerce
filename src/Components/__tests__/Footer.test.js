import React from 'react';
import { render } from '@testing-library/react';
import Footer from '../Footer';

describe('Product Container', () => {
  it('Must return entire product container', () => {
    const { asFragment } = render(<Footer />);

    expect(asFragment()).toMatchSnapshot();
  });
});
