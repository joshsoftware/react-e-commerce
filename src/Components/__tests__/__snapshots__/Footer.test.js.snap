// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Container Must return entire product container 1`] = `
<DocumentFragment>
  <footer
    class="bg-dark text-white"
  >
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col-4"
        >
          <ul
            class="list-inline"
          >
            <li
              class="rounded-sm"
            >
              Categories
            </li>
            <li
              class="rounded-sm"
            >
              <a
                href="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png"
              >
                About Us
              </a>
            </li>
            <li
              class="rounded-sm"
            >
              <a
                href="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png"
              >
                Testomonials
              </a>
            </li>
            <li
              class="rounded-sm"
            >
              <a
                href="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png"
              >
                Contact
              </a>
            </li>
            <li
              class="rounded-sm"
            >
              <a
                href="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png"
              >
                Journal
              </a>
            </li>
            <li
              class="rounded-sm"
            >
              <a
                href="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png"
              >
                Privacy Policy
              </a>
            </li>
          </ul>
        </div>
        <div
          class="col-4"
        >
          <ul
            class="list-inline"
          >
            <li
              class="rounded-sm"
            >
              Partners
            </li>
            <li
              class="rounded-sm"
            >
              Support
            </li>
            <li
              class="rounded-sm"
            >
              Shipping & Returns
            </li>
            <li
              class="rounded-sm"
            >
              Size Guide
            </li>
            <li
              class="rounded-sm"
            >
              Product Care
            </li>
          </ul>
        </div>
        <div
          class="col-4"
        >
          <ul
            class="list-inline"
          >
            <li
              class="rounded-sm"
            >
              Contact Us
            </li>
            <li
              class="rounded-sm"
            >
              Josh
            </li>
            <li
              class="rounded-sm"
            >
              Pune
            </li>
            <li
              class="rounded-sm"
            >
              Bavdhan
            </li>
            <li
              class="rounded-sm"
            >
              +974524379
            </li>
          </ul>
        </div>
        <div
          class="footer-bottom"
        >
          Â©Copyright XYZ.ltd 2020
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
